feat: add url check before download

burnsjared0415 opened the pull request:
<!--

In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/packer-examples-for-vsphere/blob/main/CONTRIBUTING.md) for making a pull request.

-->

# <!-- Intentionally left blank. -->

## Summary of Pull Request

Added a check to verify a download URL before starting the download of an ISO.

<!--
    Please provide a clear and concise description of the pull request.
-->

## Type of Pull Request

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [ ] This is a bugfix. `type/bug`
- [x] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.
      Please describe:

## Related to Existing Issues

<!--
  Is this related to any GitHub issue(s)?

  For example:
  - Resolves #1234
-->

Issue Number: N/A

## Test and Documentation Coverage

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [x] Tests have been completed.
- [ ] Documentation has been added or updated.

## Breaking Changes?

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

<!--
    If this pull request contains a breaking change, please describe the impact and mitigation path.
-->

burnsjared0415 committed:
feat: gos iso download automation (#874)

Adds a script (`./download.sh`) and a configuration (`project.json`) to assist in the download of the guest operating systems.

Signed-off-by: Jared Burns <jared.burns@broadcom.com>
Co-authored-by: Ryan Johnson <ryan.johnson@broadcom.com>

burnsjared0415 committed:
feat: added iso check

added ISO check before starting download. this will prevent download from starting if iso exist.

Signed-off-by: Jared Burns <jared.burns@broadcom.com>

burnsjared0415 committed:
feat: add url check before download

Added a check to verify a download URL before starting the download of an ISO.

Signed-off-by: Jared Burns <jared.burns@broadcom.com>

tenthirtyam commented on download.sh:
```diff
@@ -345,12 +345,26 @@ select_download() {
+        if curl --output /dev/null --silent --head --fail "$download_link"; then
+           download_dir="$(echo "${iso_base_path}/${os}/${dist}/${version}/${arch}" | tr '[:upper:]' '[:lower:]' | tr ' ' '-')"
+        else
+           echo "Download link $download_link  not valid. Skip download."
```
```suggestion
           print_message invalid "Download link $download_link  not valid. Please check the provided URL."
```

tenthirtyam commented on download.sh:
```diff
@@ -455,13 +477,27 @@ select_download() {
+        if curl --output /dev/null --silent --head --fail "$download_link"; then
+           download_dir="$(echo "${iso_base_path}/${os}/${dist}/${version}/${arch}" | tr '[:upper:]' '[:lower:]' | tr ' ' '-')"
+        else
+           echo "Download link $download_link  not valid. Skip download."
```
```suggestion
           print_message invalid "Download link $download_link  not valid. Please check the provided URL."
```

tenthirtyam reviewed (changes requested):
Minor nit and suggestion.

tenthirtyam committed:
fix: windows pwsh provisioner (#879)

Removes the PowerShell provisioner for Windows 11 and 10 as it's not required currently after the transition to Ansible.

Signed-off-by: Ryan Johnson <ryan.johnson@broadcom.com>

tenthirtyam committed:
Merge branch 'develop' into feat/add-url-for-file-download

Signed-off-by: Ryan Johnson <ryan.johnson@broadcom.com>
